bundles/org.openhab.binding.heos/src/main/java/org/openhab/binding/heos/internal/HeosChannelHandlerFactory.java

```diff
@@ -12,117 +12,100 @@
         if (handlerStorageMap.containsKey(channelUID)) {
             return handlerStorageMap.get(channelUID);
         } else {
-            HeosChannelHandler createdChannelHandler = createNewChannelHandler(channelUID, channelTypeUID);
-            handlerStorageMap.put(channelUID, createdChannelHandler);
-            return createdChannelHandler;
+            HeosChannelHandler handler = createNewChannelHandler(channelUID, eventListener, channelTypeUID);
+            if (handler != null) {
+                handlerStorageMap.put(channelUID, handler);
+            }
+            return handler;
         }
     }
 
-    private HeosChannelHandler createNewChannelHandler(ChannelUID channelUID, ChannelTypeUID channelTypeUID) {
+    private @Nullable HeosChannelHandler createNewChannelHandler(ChannelUID channelUID, HeosEventListener eventListener,
+            @Nullable ChannelTypeUID channelTypeUID) {
         switch (channelUID.getId()) {
             case CH_ID_CONTROL:
-                return new HeosChannelHandlerControl(bridge, api);
+                return new HeosChannelHandlerControl(eventListener, bridge);
             case CH_ID_VOLUME:
-                return new HeosChannelHandlerVolume(bridge, api);
+                return new HeosChannelHandlerVolume(eventListener, bridge);
             case CH_ID_MUTE:
-                return new HeosChannelHandlerMute(bridge, api);
-            case CH_ID_PLAY_URL:
-                return new HeosChannelHandlerPlayURL(bridge, api);
+                return new HeosChannelHandlerMute(eventListener, bridge);
             case CH_ID_INPUTS:
-                return new HeosChannelHandlerInputs(bridge, api);
-            case CH_ID_UNGROUP:
-                return new HeosChannelHandlerGrouping(bridge, api);
-            case CH_ID_RAW_COMMAND:
-                return new HeosChannelHandlerRawCommand(bridge, api);
-            case CH_ID_REBOOT:
-                return new HeosChannelHandlerReboot(bridge, api);
-            case CH_ID_BUILDGROUP:
-                return new HeosChannelHandlerBuildGroup(bridge, api);
-            case CH_ID_PLAYLISTS:
-                return new HeosChannelHandlerPlaylist(bridge, api);
+                return new HeosChannelHandlerInputs(eventListener, bridge);
             case CH_ID_REPEAT_MODE:
-                return new HeosChannelHandlerRepeatMode(bridge, api);
+                return new HeosChannelHandlerRepeatMode(eventListener, bridge);
             case CH_ID_SHUFFLE_MODE:
-                return new HeosChannelHandlerShuffleMode(bridge, api);
+                return new HeosChannelHandlerShuffleMode(eventListener, bridge);
             case CH_ID_ALBUM:
-                return new HeosChannelHandlerAlbum(bridge, api);
             case CH_ID_SONG:
-                return new HeosChannelHandlerTitle(bridge, api);
             case CH_ID_ARTIST:
-                return new HeosChannelHandlerArtist(bridge, api);
             case CH_ID_COVER:
```

Comment:
Are those cases are still required?

Reply:
Yes, they all are all handled in a single case, Java has a fall-through by default. So they all end up in: 
```java
                return new HeosChannelHandlerNowPlaying(eventListener, bridge);
```